packages/scoped-elements/src/ScopedElementsMixin.js

```diff
@@ -1,194 +1,113 @@
- * @param {Cache<string, string>} tagsCache
- * @returns {function(any): any}
- */
-const scopedElementsTemplateFactory = (
-  scopeName,
-  scopedElements,
-  templateCache,
-  tagsCache,
-) => template => {
-  const newTemplate = transformTemplate(template, scopedElements, templateCache, tagsCache);
-
-  return shadyTemplateFactory(scopeName)(newTemplate);
-};
-
-/** @type {ScopedElementsMixin} */
 const ScopedElementsMixinImplementation = superclass =>
+  /** @type {ScopedElementsMixin} */
   class ScopedElementsHost extends superclass {
     /**
-     * Obtains the scoped elements definitions map
+     * Obtains the scoped elements definitions map if specified.
      *
      * @returns {ScopedElementsMap}
      */
     static get scopedElements() {
       return {};
     }
 
-    /** @override */
-    static render(template, container, options) {
-      if (!options || typeof options !== 'object' || !options.scopeName) {
-        throw new Error('The `scopeName` option is required.');
-      }
-      const { scopeName, eventContext } = options;
-
-      const templateCache = getTemplateCache(eventContext);
-      const tagsCache = getTagsCache(eventContext);
-      const { scopedElements } = this;
-
-      return super.render(template, container, {
-        ...options,
-        templateFactory: scopedElementsTemplateFactory(
-          scopeName,
-          scopedElements,
-          templateCache,
-          tagsCache,
-        ),
-      });
+    /** @type {ShadowRootInit} */
+    static get shadowRootOptions() {
+      return this.__shadowRootOptions;
     }
 
-    /**
-     * Defines a scoped element
-     *
-     * @param {string} tagName
-     * @param {typeof HTMLElement} klass
-     */
-    defineScopedElement(tagName, klass) {
-      return defineScopedElement(tagName, klass, getTagsCache(this));
+    /** @type {ShadowRootInit} */
+    static set shadowRootOptions(value) {
+      this.__shadowRootOptions = value;
+    }
+
+    /** @type {CSSResultFlatArray} */
+    static get elementStyles() {
+      return this.__elementStyles;
+    }
+
+    static set elementStyles(styles) {
+      this.__elementStyles = styles;
+    }
+
+    // either TS or ESLint will complain here
+    // eslint-disable-next-line no-unused-vars
+    constructor(..._args) {
+      super();
+      /** @type {RenderOptions} */
+      this.renderOptions = this.renderOptions || undefined;
     }
 
     /**
-     * Returns a scoped tag name
+     * Obtains the CustomElementRegistry if specified.
      *
-     * @deprecated Please, use the instance method instead of the static one. This static method is not able to
-     * obtain the tagName of lazy defined elements, while the instance one is.
-     * @param {string} tagName
-     * @returns {string|undefined}
+     * @return {CustomElementRegistry}
      */
-    static getScopedTagName(tagName) {
-      // @ts-ignore
-      const klass = this.scopedElements[tagName];
+    get registry() {
+      if (!this._registry) {
```

Comment:
Should we attach the registry to the class by default?

Reply:
=> will tackle this in another PR (before we release)